pkg/plugins/scms/gitea/main.go

```diff
@@ -114,6 +114,8 @@ type Spec struct {
 			If you need to push changes to a specific branch, you must use the plugin "git" instead of this
 	*/
 	Branch string `yaml:",omitempty"`
+	// Whether to checkout submodules: `true` to checkout submodules or `false` to skip.
+	Submodules *bool `yaml:",omitempty"`
```

Comment:
I really like this approach of using a pointer to control if the default value must be false or true